flink-python/pyflink/table/table_result.py

```diff
@@ -48,6 +51,22 @@ def get_job_client(self):
         else:
             return None
 
+    def wait(self, timeout_ms=None):
+        """
+        Wait if necessary for at most the given time (milliseconds) for the data to be ready.
+
+        For select operation, this method will wait unit the first row can be accessed in local.
```

Comment:
For a select operation, this method will wait until the first row can be accessed locally.
For an insert operation, this method will wait for the job to finish, because the result contains only one row.
For other operations, this method will return immediately, because the result is already available locally.